lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -2649,6 +2649,114 @@ def calc_arrow(uvw, angle=15):
 
     quiver3D = quiver
 
+    def voxels(self, filled, color=None):
+        """
+        Plot a set of filled voxels
+
+        All voxels are plotted as 1x1x1 cubes on the axis, with filled[0,0,0]
+        placed with its lower corner at the origin. Occluded faces are not plotted
+
+        Parameters
+        ----------
+        filled : np.array
+            A 3d array of values, with truthy values indicating which voxels
+            to fill
+
+        color : array_like
+            Either a single value or an array the same shape as filled,
+            indicating what color to draw the faces of the voxels. If None,
+            plot all voxels in the same color, the next in the color sequence
+
+        Returns
+        -------
+        list of `Poly3DCollection`
+            The square faces which were plotted, in an arbitrary order
+        """
+        # check dimensions, and deal with a single color
+        if filled.ndim != 3:
+            raise ValueError("Argument filled must be 3-dimensional")
+
+        if color is None:
+            color = next(self._get_patches_for_fill.prop_cycler)['color']
+        if np.ndim(color) <= 1:
+            color, _ = np.broadcast_arrays(
+                color,
+                filled[np.index_exp[...] + np.index_exp[np.newaxis] * np.ndim(color)]
```

Comment:
Since numpy/numpy@f897e1fc50d41e219b0868b1394dee9b118082d0 back in 2002, so I assume this is fine. It's essentially the same as `np.s_[...]`, but returns a tuple